components/post_view/reaction/reaction.jsx

```diff
@@ -10,88 +10,152 @@ import OverlayTrigger from 'components/overlay_trigger';
-
-        /*
-         * The name of the emoji for the reaction
-         */
         emojiName: PropTypes.string.isRequired,
-
-        /*
-         * The number of reactions to this post for this emoji
-         */
         reactionCount: PropTypes.number.isRequired,
-
-        /*
-         * Array of users who reacted to this post
-         */
         profiles: PropTypes.array.isRequired,
-
-        /*
-         * The number of users not in the profile list who have reacted with this emoji
-         */
         otherUsersCount: PropTypes.number.isRequired,
-
-        /*
-         * Array of reactions by user
-         */
         reactions: PropTypes.arrayOf(PropTypes.object).isRequired,
-
-        /*
-         * True if the user has the permission to add a reaction in this channel
-         */
         canAddReaction: PropTypes.bool.isRequired,
-
-        /*
-         * True if user has the permission to remove his own reactions in this channel
-         */
         canRemoveReaction: PropTypes.bool.isRequired,
-
-        /*
-         * The URL of the emoji image
-         */
         emojiImageUrl: PropTypes.string.isRequired,
-
         actions: PropTypes.shape({
-
-            /*
-             * Function to add a reaction to a post
-             */
             addReaction: PropTypes.func.isRequired,
-
-            /*
-             * Function to get non-loaded profiles by id
-             */
             getMissingProfilesByIds: PropTypes.func.isRequired,
-
-            /*
-             * Function to remove a reaction from a post
-             */
             removeReaction: PropTypes.func.isRequired,
         }),
         sortedUsers: PropTypes.object.isRequired,
     }
 
-    handleAddReaction = (e) => {
-        e.preventDefault();
-        const {actions, post, emojiName} = this.props;
-        actions.addReaction(post.id, emojiName);
+    constructor(props) {
+        super(props);
+
+        const {reactionCount, canRemoveReaction, canAddReaction} = this.props;
+        const {currentUserReacted} = this.props.sortedUsers;
+
+        if (currentUserReacted) {
+            this.state = {
+                userReacted: currentUserReacted,
+                reactedClass: 'Reaction reacted',
+                displayNumber: reactionCount,
+                reactedNumber: reactionCount,
+                unreactedNumber: reactionCount - 1,
+            };
+        } else {
+            this.state = {
+                userReacted: currentUserReacted,
+                reactedClass: 'Reaction unreacted',
+                displayNumber: reactionCount,
+                reactedNumber: reactionCount + 1,
+                unreactedNumber: reactionCount,
+            };
+        }
+
+        if (currentUserReacted && !canRemoveReaction) {
+            this.state = {
+                ...this.state,
+                reactedClass: 'Reaction read-only',
+            };
+        } else if (!currentUserReacted && !canAddReaction) {
```

Comment:
this can be simplified `!(currentUserReacted || canAddReaction)`

Reply:
You’re right, thanks